prospector/client/cli/prospector_client.py

```diff
@@ -59,88 +58,87 @@ def prospector(  # noqa: C901
     limit_candidates: int = MAX_CANDIDATES,
     rules: "list[str]" = ["ALL"],
     model_name: str = "",
-    console_msg_func: Callable = msg,
```

Comment:
this allowed to inject some console writer other then the default one (e.g., for testing purposes)

Reply:
I thought about this, but shouldn't achieve this with mocking (for example https://pypi.org/project/pytest-mock/), instead of polluting production code with test-related stuff?

Reply:
Testability is a positive side-effect, not necessarily the main rationale for this, which is essentially modularity achieved through IoC: this makes explicit the dependency between the prospector client and the component through which it communicates with the user. Arguably, this is not very elegant nor expressive because the contract is very loosely specified (it could be improved by defining a specific subclass of Callable, but I did not want to add more code than strictly necessary at this point).

If we drop this parameter, the dependency remains, but it is just implicit.

Did I convince you? ;-)

Reply:
sort of... where do you draw the line? you did not introduce DI for file or network reading, but you wish to include it for console writes?